src/lavinmq/shovel/shovel.cr

```diff
@@ -382,7 +399,15 @@ module LavinMQ
           @retries = 0
           @source.each do |msg|
             @message_count += 1
-            @destination.push(msg, @source)
+            push_retries = 0
+            until @destination.push(msg, @source)
+              if push_retries >= 5
+                @source.reject(msg.delivery_tag)
```

Comment:
Wasn't the message rejected with requeue? that means the message will be retried soon again?

Reply:
The idea is to try 5 times and then put the message back in the queue so it could be removed or consumed by another consumer. 
Don't want to reject without requeue since the message will disappear if no DLX is configured and the issue related to this PR #1357 was just about disappearing messages.